mutable_tree.go

```diff
@@ -673,8 +677,8 @@ func (tree *MutableTree) GetVersioned(key []byte, version int64) []byte {
 				return nil
 			}
 
-			if fastNode != nil && fastNode.versionLastUpdatedAt <= version {
-				return fastNode.value
+			if fastNode != nil && fastNode.Version() <= version {
```

Comment:
Please add a comment above fastNode's method implementation that this is the version it was last updated at